test/parallel/test-tls-set-secure-context.js

```diff
@@ -19,54 +23,63 @@ const credentialOptions = [
     ca: fixtures.readKey('ca2-cert.pem')
   }
 ];
-let requestsCount = 0;
 let firstResponse;
 
 const server = https.createServer(credentialOptions[0], (req, res) => {
-  requestsCount++;
+  const id = +req.headers.id;
 
-  if (requestsCount === 1) {
+  if (id === 1) {
     firstResponse = res;
     firstResponse.write('multi-');
     return;
-  } else if (requestsCount === 3) {
+  } else if (id === 4) {
     firstResponse.write('success-');
   }
 
   res.end('success');
 });
 
-server.listen(0, common.mustCall(async () => {
+server.listen(0, common.mustCall(() => {
   const { port } = server.address();
-  const firstRequest = makeRequest(port);
+  const firstRequest = makeRequest(port, 1);
+
+  async function makeRemainingRequests() {
+    // Wait until the first request is guaranteed to have been handled.
+    if (!firstResponse) {
+      return setImmediate(makeRemainingRequests);
```

Comment:
Can we use a dummy EventEmitter to emit an event when `firstResponse` is set instead of polling for it and call `makeRemainingRequests` when the event is emitted?